pkg/ui/react-app/src/thanos/pages/blocks/TimeRange.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC, useMemo } from 'react';
+import Slider from 'rc-slider';
+import 'rc-slider/assets/index.css';
+import moment from 'moment';
+import styles from './blocks.module.css';
+
+const Range = Slider.Range;
+
+interface TimeRangeProps {
+  viewMinTime: number;
+  viewMaxTime: number;
+  gridMinTime: number;
+  gridMaxTime: number;
+  onChange: React.Dispatch<React.SetStateAction<[number, number]>>;
+}
+
+const TimeRange: FC<TimeRangeProps> = ({ viewMinTime, viewMaxTime, gridMinTime, gridMaxTime, onChange }) => {
+  const marks = useMemo(() => {
+    const NUM_MARKS = 10;
```

Comment:
This should likely live outside of the function as it doesn't depend on the func running